vortex-expr/src/encoding.rs

```diff
@@ -0,0 +1,95 @@
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-FileCopyrightText: Copyright the Vortex contributors
+
+use std::any::Any;
+use std::fmt::{Debug, Display, Formatter};
+
+use arcref::ArcRef;
+use vortex_array::DeserializeMetadata;
+use vortex_error::{VortexExpect, VortexResult};
+
+use crate::{ExprRef, IntoExpr, VTable};
+
+pub type ExprId = ArcRef<str>;
+pub type ExprEncodingRef = ArcRef<dyn ExprEncoding>;
```

Comment:
The encoding for an array made a logical sense in my mind, encoding for expression is very forced analogy. We can rename it later